File: JavaScript/프로그래머스/0/120848. 팩토리얼/팩토리얼.js
function solution(n) {
    let dp = new Array(10).fill(0)
    dp[0] = 0
    dp[1] = 1
    
    for(let i = 2 ; i < 11 ; i++) {
        dp[i] = dp[i-1] * i
    }
    
    return dp.filter(v=> v<=n).length-1
}

File: Kotlin/프로그래머스/0/120848. 팩토리얼/팩토리얼.kt
class Solution {
    fun fact(n: Int):Int {
        if (n == 1) return 1
        return fact(n-1) * n
    }
    
    fun solution(n: Int): Int {
        var answer: Int = 0
        
        for (i in 10 downTo 1) {
            if(fact(i) <= n) {
                answer = i
                break
            }
        }
        
        return answer
    }
}